python-oauth-web-app/atproto_oauth.py

```diff
@@ -336,6 +338,24 @@ def pds_dpop_jwt(
     return dpop_proof
 
 
+# A resource server may signal the need for a [new] DPoP nonce via one of two methods
+# 1. WWW-Authenticate header with paramater error="use_dpop_nonce" (see https://datatracker.ietf.org/doc/html/rfc9449#RSNonce)
+# 2. JSON response body with field error="use_dpop_nonce"
+# The latter is only supposed to be returned by an Authorization Server (see https://datatracker.ietf.org/doc/html/rfc9449#name-authorization-server-provid), but we support it anyway.
+def is_use_dpop_nonce_error_response(resp: Response):
```

Comment:
reasonable way to handle this (just check for either mechanism on every response)